### PR TITLE
let app and upgrade command accept an AppVersion uuid as target version

### PR DESCRIPTION
**Pull request by niristius**
this is my implementation to realize the requested feature from https://github.com/mittwald/cli/issues/922

### CODE REVIEW

**Comment by niristius** [comment]
upgrades are included too

File: src/lib/resources/app/Installer.tsx [hunk 1]
```diff
@@ -91,11 +94,29 @@ export class AppInstaller<TFlagName extends AvailableFlagName> {
...
+    var appVersion: AppAppVersion;
+
+    if (validateUuid(appVersionId)) {
+      appVersion = await getAppVersionFromUuid(
+        apiClient,
+        this.appId,
+        appVersionId,
+      );
+    } else {
+      appVersion = await normalizeToAppVersionUuid(
```
Comment (martin-helmich) [inline src/lib/resources/app/Installer.tsx:108]: Maybe build the "if it looks like a UUID, just use that"-logic directly into `normalizeToAppVersionUuid`? IIRC, all other `normalize<whatever>` functions that we have floating around work more or less similar, anyway.

File: src/commands/app/upgrade.tsx [hunk 2]
```diff
@@ -114,7 +115,19 @@ export class UpgradeApp extends ExecRenderBaseCommand<typeof UpgradeApp, void> {
...
+    ) {
+      targetAppVersion = await getAppVersionFromUuid(
+        this.apiClient,
+        currentApp.id,
+        this.flags["target-version"],
+      );
+    } else if (
+      this.flags["target-version"] &&
+      !validateUuid(this.flags["target-version"])
+    ) {
```
Comment (martin-helmich) [inline src/commands/app/upgrade.tsx:130]: These `AND`ed conditions could be deconstructed to (maybe?) more readable nested conditions:

```
else if (this.flags["target-version"]) {
  if (validateUuid(this.flags["target-version"])) {
    // ...
  } else {
    // ...
  }
} // ...
```

Also, the way how `targetAppVersion` is getting more and more complex, with more and more branches in this condition. I would suggest extracting the entire logic of populating `targetAppVersion` into a function.

**Review by martin-helmich — CHANGES_REQUESTED** [review]
Some refactoring suggetions; also, mind the linter issues.

**Pull request merged**